public/app/features/explore/ExploreToolbar.tsx

```diff
@@ -152,6 +155,16 @@ export class UnConnectedExploreToolbar extends PureComponent<Props> {
     return datasourceName ? exploreDatasources.find(datasource => datasource.name === datasourceName) : undefined;
   };
 
+  copyAndSaveShortLink = async () => {
+    const shortLink = await createShortLink(window.location.href);
+    if (shortLink) {
+      copyStringToClipboard(shortLink);
+      appEvents.emit(AppEvents.alertSuccess, ['Shortened link copied to clipboard']);
```

Comment:
nit: As Explore uses Redux I think you should dispatch the `notifyApp` action to create these application wide messages. Maybe open a separate PR that remove all usages in Explore and remove emitter from state?

Reply:
Great nit Hugo. I'll create an issue for this.

Reply:
https://github.com/grafana/grafana/issues/28262